File: A1-q2.py
#program to print the factorial of a number
n = -1

while (n<0):
    x= input("Please enter a number:\n")
    try:
        n=int(x);
    except ValueError:
        print( "{input} invalid. Please enter a NUMBER.\n".format(input=x))

if n==0:
    print("The factorial of the number", n, "is 1\n")

else:
    p=n
    fac = 1
    while (n > 0):
        fac = fac * n
        n -= 1
    print("The factorial of the number",p, "is", fac ,"\n")

File: A1-q2.c
//program to print the factorial of a number

#include <stdio.h>
void main()
{
	int n, fac = 1, i = 1;
	printf("Please enter a number:\n");
	scanf_s("%d", &n);

	if (n < 0)
	{
		do
		{
			printf("Please enter a valid number:\n");
			scanf_s("%d", &n);
		} while (n < 0);
	}

	while (n > 0 && i <= n)
	{
		fac = fac * i;
		i++;
	}

	printf("The factorial for the number %d is: %d \n", n, fac);
	return 0;
}
